composition/api/v1alpha1/composition_types.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ConditionType defines the type of ManagedConfigSync condition
+type ConditionType string
+
+// The valid conditions of Compositions
+const (
+	Ready ConditionType = "Ready"
+	// Error implies the last reconcile attempt failed
+	Error ConditionType = "Error"
```

Comment:
`Error` is a catch all for any error (temporary or otherwise) when we reconcile a composition. This may be temporary errors such as unable to reach k8s server, grpc server etc. We have a separate `Waiting` condition value to indicate we are waiting for something (we are not done yet).